osu.Game/Online/Multiplayer/IMultiplayerRoomServer.cs

```diff
@@ -40,6 +41,13 @@ public interface IMultiplayerRoomServer
         /// <exception cref="NotJoinedRoomException">If the user is not in a room.</exception>
         Task ChangeState(MultiplayerUserState newState);
 
+        /// <summary>
+        /// Change the user's local availability state of the beatmap set in joined room.
+        /// This will also force user state back to <see cref="MultiplayerUserState.Idle"/>.
+        /// </summary>
+        /// <param name="newBeatmapAvailability">The proposed new beatmap availability state.</param>
+        Task ChangeBeatmapAvailability(BeatmapAvailability newBeatmapAvailability);
```

Comment:
how will the user signal that they're done downloading? will they send a `Downloading` message with a `100%` progress? not sure about that design

Reply:
I didn't go through the downloading logic we have to check for that because it's pretty much unnecessary, `100%` `Downloading` means the user downloading progress has reached 100%, `Downloaded` will mean it finished downloading and is in the importing process.